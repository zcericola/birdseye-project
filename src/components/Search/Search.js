import React from 'react';
import Grid from '@material-ui/core/Grid';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField from '@material-ui/core/TextField';
import SearchIcon from '@material-ui/icons/Search';
import SettingsIcon from '@material-ui/icons/Settings';
import useStyles from './searchStyles';
import logo from '../../assets/birdseye_logo.png';

const Search = ( props ) => {
    const {
        searchContainer
        , formField
        , birdsEyeLogo
        , icon
        , settings
        , upgradeMsg
        , upgradeLink
        , redDot

    } = useStyles();
    return (
        <Grid
            item
            className={searchContainer}
            xs={12}
            md={8}
        >
            <img src={logo}
                className={birdsEyeLogo}
                alt='Birdseye logo'
            />
            <TextField
                variant='outlined'
                className={formField}
                InputProps={{
                    startAdornment: (
                        <InputAdornment position="start">
                            <SearchIcon className={icon} />
                        </InputAdornment>
                    )
                }} />
            <Grid item className={settings}>
                <span
                    className={upgradeMsg}
                >
                    10 days left
                </span >
                <span
                    className={upgradeLink}
                >
                    Upgrade
                </span>
                <SettingsIcon className={icon} />
                <div className={redDot}></div>


            </Grid>
        </Grid>



    );

};

export default Search;