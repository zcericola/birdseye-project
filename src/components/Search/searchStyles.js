import { makeStyles } from '@material-ui/styles';
import {
    defaultTextFieldStyle
    , primaryColor
    , secondaryColor
    , searchSpanText
} from '../../globalStyles';

const useStyles = makeStyles( {
    searchContainer: {
        maxHeight: 100
        , height: '90%'
        , display: 'flex'
        , alignItems: 'center'
        , background: primaryColor
        , [ '@media(maxWidth: 780px)' ]: {
            order: 1
        }

    }
    , birdsEyeLogo: {
        height: 30
        , width: 40
        , marginLeft: 20
    }
    , formField: {
        ...defaultTextFieldStyle
        , width: '50%'
        , background: secondaryColor
        , margin: '10px 0 10px 20px'
        , [ `& multilineColor` ]: {
            color: 'white'

        }

    }
    , icon: {
        fill: 'white'
    }
    , settings: {
        display: 'flex'
        , justifyContent: 'flex-end'
        , width: '40%'
    }
    , upgradeMsg: {
        ...searchSpanText
        , fontStyle: 'italic'
    }
    , upgradeLink: {
        ...searchSpanText
        , fontWeight: '700'
        , textDecoration: 'underline'
        , cursor: 'pointer'
    }
    , redDot: {
        position: 'relative'
        , borderRadius: 100
        , background: '#DC143C'
        , height: 5
        , width: 5
        , top: 16
        , right: 6
    }


} );

export default useStyles;
