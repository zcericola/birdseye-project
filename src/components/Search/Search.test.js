import React from 'react';
import ReactDOM from 'react-dom';
import { mount, debug } from 'enzyme';
import renderer from 'react-test-renderer';
import Grid from '@material-ui/core/Grid';
import Search from './Search';
import '../../setup';

describe('Search component', () => {
    it('Search renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Search />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('matches the snapshot', () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();

    });
    it('Has a Search container', () => {
        const wrapper = mount(<Search />);

        const container = wrapper.find(Grid);
        expect(container).toEqual({});
        wrapper.unmount();
    });

});
