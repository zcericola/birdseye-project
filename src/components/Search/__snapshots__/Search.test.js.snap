// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component matches the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-searchContainer-163 MuiGrid-item"
>
  <img
    alt="Birdseye logo"
    className="makeStyles-birdsEyeLogo-164"
    src="birdseye_logo.png"
  />
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-formField-165"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-309 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-310"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root makeStyles-icon-166"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-settings-167 MuiGrid-item"
  >
    <span
      className="makeStyles-upgradeMsg-168"
    >
      10 days left
    </span>
    <span
      className="makeStyles-upgradeLink-169"
    >
      Upgrade
    </span>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root makeStyles-icon-166"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h20v20H0V0z"
        fill="none"
        transform="scale(1.2, 1.2)"
      />
      <path
        d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
        transform="scale(1.2, 1.2)"
      />
    </svg>
    <div
      className="makeStyles-redDot-170"
    />
  </div>
</div>
`;
